### PR TITLE
Fix re-exports in Ember >= v4.0.0-beta.9

### LINKED ISSUE #5: Import path does not work in v0.3.1?
With recent changes in Ember beta, looks like those imports from `README.md` don't work any more:

```js
import {
  Checkbox,
  LinkComponent,
  TextArea,
  TextField
} from '@ember/legacy-built-in-components';
```

[The test app itself uses](https://github.com/emberjs/ember-legacy-built-in-components/blob/v0.3.1/tests/dummy/app/components/legacy-checkbox.js) imports like
```js
export { default } from '@ember/legacy-built-in-components/components/checkbox';
```

The problem comes from the `addon/index.js` file having this snippet:
```js
export const Checkbox = macroCondition(
  dependencySatisfies('ember-source', '>= 3.27.0-beta.1')
)
  ? Ember._LegacyCheckbox.extend()
  : Ember.Checkbox;
```

Unless I'm missing something, `Ember._Legacy*` have been removed in https://github.com/emberjs/ember.js/commit/5d24ae7480aae886d7a526d15332f53fe5b1689c, specifically those lines

```js
// Expose a non-deprecated version for tests and the @ember/legacy-built-in-components addon
Ember[`_Legacy${name}`] = value;
```

@mixonic @chancancode what do you think would be the right fix here? I'm thinking of two options:

1. replace `README.md` import example to (breaking change to addon contract)
```js
export Checkbox from '@ember/legacy-built-in-components/components/checkbox';
```
2. Fix code in `addon/index.js` to be smth like:
```js
macroCondition(dependencySatisfies('ember-source', '>= 3.27.0-beta.1'))
  ? macroCondition(dependencySatisfies('ember-source', '< v4.0.0-beta.9'))
    ? Ember._LegacyCheckbox.extend()
    : importSync('@ember/legacy-built-in-components/components/checkbox')
  : Ember.Checkbox;
```

### PR DESCRIPTION
**Pull request by SergeAstapov**
Fixes #5

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
+  dependencySatisfies,
+  importSync,
+} from '@embroider/macros';
 
-export const Checkbox = macroCondition(
-  dependencySatisfies('ember-source', '>= 3.27.0-beta.1')
-)
-  ? Ember._LegacyCheckbox.extend()
-  : Ember.Checkbox;
+let Checkbox;
```
Comment (SergeAstapov) [inline addon/index.js:9]: Maybe my personal preference, but nested ternary operators hard to read.

Another option could be to do smth like:
```js
export const Checkbox = macroCondition(
  dependencySatisfies('ember-source', '>= 3.27.0-beta.1')
)
  ? importSync('@ember/legacy-built-in-components/components/checkbox').default
  : Ember.Checkbox;
```
but I wasn't confident as this addon was using `Ember._LegacyCheckbox.extend()` for 3.27 and 3.28

**Comment by oliverlangan** [comment]
We were having this same issue in one of our addons, and this approach works to address it. The _types_ get all messed up in Typescript, but we can deal with that.

**Comment by mixonic** [comment]
Excellent catch, thank you. I'll merge.

I do think an alternative is to *always* export the components vendored into this addon. There is no need to re-export the ones from Ember's code now that we have copied them over.

**Comment by RobbieTheWagner** [comment]
@SergeAstapov in Ember 4+ I am getting `undefined` for all the imports. Should this be working?

**Comment by SergeAstapov** [comment]
@rwwagner90 yes, it should work. I'll check if anything has changed

**Comment by RobbieTheWagner** [comment]
@SergeAstapov it seems there is a conflict of any addons use embroider 0.50+. We'll need to update our version in this addon and release a breaking change.

**Comment by SergeAstapov** [comment]
@rwwagner90 raised #11 - not sure we actually need a major version bump as technically for users of this addons things should not have changed

**Pull request merged**